### PR TITLE
Entitlement api fixes

### PR DESCRIPTION
**Pull request by opqdonut**
discovered as part of #1711 

could consider breaking backwards compatibility also, see TODO comment

# Definition of Done / Review checklist

## Reviewability
- [x] link to issue

## API
- [x] API is documented and shows up in Swagger UI
- [x] API is backwards compatible or completely new

## Documentation
- [x] update changelog if necessary

## Testing
- [x] valuable features are integration / browser / acceptance tested automatically

### CODE REVIEW

File: src/clj/rems/api/schema.clj [hunk 1]
```diff
@@ -51,6 +60,7 @@
 
 (s/defschema Entitlement
   {:resource s/Str
+   :userid UserId ;; TODO could have UserWithAttributes here instead of :userid and :mail
```
Comment (luontola) [inline src/clj/rems/api/schema.clj]: 👍

Comment (luontola) [reply src/clj/rems/api/schema.clj]: For a transition period we could have both.

File: src/clj/rems/db/entitlements.clj [hunk 2]
```diff
@@ -27,7 +28,7 @@
         (db/get-entitlements {:user (if (has-roles? :handler :owner :reporter)
                                       user-or-nil
                                       (getx-user-id))
-                              :resource resource-or-nil
+                              :resource-ext-id resource-or-nil
```
Comment (luontola) [inline src/clj/rems/db/entitlements.clj]: How did the tests pass before this change? 😟

File: src/clj/rems/api/schema.clj (hunk 1 above)
Comment (opqdonut) [reply src/clj/rems/api/schema.clj]: hmm okay :userid is fresh so I can replace it with :user, and keep :mail with a deprecation comment

**Pull request merged**